libraries/AP_RCTelemetry/AP_CRSF_Telem.cpp

```diff
@@ -1181,60 +1236,122 @@ void AP_CRSF_Telem::calc_parameter() {
+        _telem.ext.param_entry.header.chunks_left = 0;  // assume menus fit in a single chunk
+
+        _telem.ext.param_entry.payload[idx++] = menu->parent_id; // parent folder
+        _telem.ext.param_entry.payload[idx++] = ParameterType::FOLDER; // type
+        // name
+        strncpy((char*)&_telem.ext.param_entry.payload[idx], menu->name, 16);
+        idx += strnlen(menu->name, 16);
+        _telem.ext.param_entry.payload[idx++] = 0; // null terminator
+
+        // menu entries of parameter ids
+        for (uint8_t i = 0; i < menu->num_params; i++) {
+            _telem.ext.param_entry.payload[idx++] = menu->params[i].id;
+        }
+        _telem.ext.param_entry.payload[idx] = 0xFF; // terminator
+
+        _telem_size = sizeof(AP_CRSF_Telem::ParameterSettingsEntryHeader) + 1 + idx;
+        _telem_type = AP_RCProtocol_CRSF::CRSF_FRAMETYPE_PARAMETER_SETTINGS_ENTRY;
+        _pending_request.frame_type = 0;
+        _telem_pending = true;
+        return;
+    }
+
+    // scripted menu parameter request indexed after scripted menu ids
+    AP_CRSF_Telem::ScriptedParameter* param = scripted_menus.find_parameter(_param_request.param_num);
+    debug("param request: %d -> 0x%p", _param_request.param_num, param);
+
+    if (param != nullptr) {
+        // find the parameter to write
+
+        _telem.ext.param_entry.header.param_num = _param_request.param_num;
+        const uint8_t CHUNK_SIZE = 56;
+        const uint8_t chunks = ((param->length - 1) / CHUNK_SIZE) + 1;
+        _telem.ext.param_entry.header.chunks_left = (chunks - 1) - _param_request.param_chunk;
+        _telem.ext.param_entry.payload[idx++] = param->parent_id; // parent folder
+
+        if (_pending_request.frame_type == AP_RCProtocol_CRSF::CRSF_FRAMETYPE_PARAMETER_READ) {
+            // payload encoded from lua
+            memcpy((uint8_t*)&_telem.ext.param_entry.payload[idx],
+                    &param->data[_param_request.param_chunk * CHUNK_SIZE],
+                    _telem.ext.param_entry.header.chunks_left > 0 ? CHUNK_SIZE : param->length % CHUNK_SIZE);
+            idx += param->length;
+        } else {
+            memcpy((uint8_t*)&_telem.ext.param_entry.payload[idx], _param_request.payload.payload,
+                    _param_request.payload.payload_length);
+            idx += _param_request.payload.payload_length;
+        }
+        _telem_size = sizeof(AP_CRSF_Telem::ParameterSettingsEntryHeader) + 1 + idx;
+        _telem_type = AP_RCProtocol_CRSF::CRSF_FRAMETYPE_PARAMETER_SETTINGS_ENTRY;
+        _pending_request.frame_type = 0;
+        _telem_pending = true;
+
+        return;
+    }
+#endif // AP_CRSF_SCRIPTING
+
     AP_OSD* osd = AP::osd();
 
     if (osd == nullptr) {
         return;
     }
 
-    AP_OSD_ParamSetting* param = osd->get_setting((_param_request.param_num - 1) / AP_OSD_ParamScreen::NUM_PARAMS,
+    AP_OSD_ParamSetting* setting = osd->get_setting((_param_request.param_num - 1) / AP_OSD_ParamScreen::NUM_PARAMS,
```

Comment:
Pull this name change out as a separate PR and we can get it merged straight away...

Reply:
The name change is because there is another variable with the same name and they need to be different